packages/flutter/lib/src/animation/curves.dart

```diff
@@ -267,6 +304,629 @@ class Cubic extends Curve {
+/// smoothly from one control point to the next, passing through the control
+/// points.
+///
+/// {@template flutter.animation.curves.catmull_rom_description}
+/// Unlike most cubic splines, Catmull-Rom splines have the advantage that their
+/// curves pass through the control points given to them. They are cubic
+/// polynomial representations, and, in fact, Catmull-Rom splines can be
+/// converted mathematically into cubic splines. This class implements a
+/// "centripetal" Catmull-Rom spline. The term centripetal implies that it won't
+/// form loops or self-intersections within a single segment.
+/// {@endtemplate}
+///
+/// See also:
+///  * [Centripetal Catmull–Rom splines](https://en.wikipedia.org/wiki/Centripetal_Catmull%E2%80%93Rom_spline)
+///    on Wikipedia.
+///  * [Parameterization and Applications of Catmull-Rom Curves](http://faculty.cs.tamu.edu/schaefer/research/cr_cad.pdf),
+///    a paper on using Catmull-Rom splines.
+///  * [CatmullRomCurve], an animation curve that uses a [CatmullRomSpline] as its
+///    internal representation.
+class CatmullRomSpline extends Curve2D {
+  /// Constructs a centripetal Catmull-Rom spline curve.
+  ///
+  /// The `controlPoints` argument is a list of four or more points that
+  /// describe the points that the curve must pass through.
+  ///
+  /// The optional `tension` argument controls how tightly the curve approaches
+  /// the given `controlPoints`. It must be in the range 0.0 to 1.0, inclusive. It
+  /// defaults to 0.0, which provides the smoothest curve. A value of 1.0
+  /// produces a linear interpolation between points.
+  ///
+  /// The optional `endHandle` and `startHandle` points are the beginning and
+  /// ending handle positions. If not specified, they are created automatically
+  /// by extending the line formed by the first and/or last line segment in the
+  /// `controlPoints`, respectively. The spline will not go through these handle
+  /// points, but they will affect the slope of the line at the beginning and
+  /// end of the spline. The spline will attempt to match the slope of the line
+  /// formed by the start or end handle and the neighboring first or last
+  /// control point. The default is chosen so that the slope of the line at the
+  /// ends matches that of the first or last line segment in the control points.
+  ///
+  /// The `tension` and `controlPoints` arguments must not be null, and the
+  /// `controlPoints` list must contain at least four control points to
+  /// interpolate.
+  ///
+  /// The internal curve data structures are lazily computed the first time
+  /// [transform] is called.  If you would rather pre-compute the structures,
+  /// use [CatmullRomSpline.precompute] instead.
+  CatmullRomSpline(
+      List<Offset> controlPoints, {
+        double tension = 0.0,
+        Offset startHandle,
+        Offset endHandle,
+      }) : assert(controlPoints != null),
+        assert(tension != null),
```

Comment:
Indentation is still off here :)

Reply:
Argh.  That was IntelliJ "helping" me when I copy/pasted...  Fixed.